greedy_algorithms/kruskals_minimum_spanning_tree.cpp

```diff
@@ -50,20 +57,132 @@ void findMinimumEdge(const int &infinity,
 }
 }  // namespace greedy_algorithms
 
+/**
+ * define a large constant value for int
+ * define a large constant value for float
+ * define a large constant value for double
+ * define infinity for uint32_t
+ */
+constexpr int INFINITY_INT = std::numeric_limits<int>::max();
+constexpr float INFINITY_FLOAT = std::numeric_limits<float>::max();
+constexpr double INFINITY_DOUBLE = std::numeric_limits<double>::max();
+constexpr uint32_t INFINITY_UINT32 = UINT32_MAX;
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    // Test case with interger values
+    std::cout << "\nTest Case 1 :\n";
+    std::array<std::array<int, 6>, 6> graph1{
+             0,            4,            1,             4,        INFINITY_INT,   INFINITY_INT,
+             4,            0,            3,             8,             3,         INFINITY_INT,
+             1,            3,            0,        INFINITY_INT,       1,         INFINITY_INT,
+             4,            8,       INFINITY_INT,        0,            5,              7,
```

Comment:
could you follow the same formatting for the rest of the test cases as well?

Reply:
I changed it. Can you review it?